### PR TITLE
Improve home page UI.

### PR DESCRIPTION
**Pull request by gauravsingh5**
Made the UI of the home page a little better.

### CODE REVIEW

**Comment by gauravsingh5** [comment]
Site isn't responsive on Netlify, but when I run it locally it is.

**Comment by gauravsingh5** [comment]
I don't know why this is happening but whenever I open it in mobile view it doesn't render correctly until I click on a card and then go back to the home page.

**Comment by gauravsingh5** [comment]
It's changing after building. So it's not a Netlify problem

**Comment by gauravsingh5** [comment]
I overthought it. It was really simple. I didn't even need react-device-detect.

**Comment by ming-tsai** [comment]
Hi @gauravsingh5,
Thanks for contributions, could your replace double quotes to single?
I should configure prettier

**Comment by ming-tsai** [comment]
Hi @gauravsingh5,
Please ensure the card is the same size and the ArrowForward button should on left side
image

**Review by ming-tsai — CHANGES_REQUESTED** [review]
Sorry, and rollback the change of hiding navbar, when has a lot of dungeons, or in the future, we will use it

File: src/pages/404.jsx [hunk 1]
```diff
@@ -1,6 +1,6 @@
 import React from 'react';
-import Layout from '../components/layout';
 import { Link } from 'gatsby';
+import Layout from '../components/layout';
```
Comment (ming-tsai) [inline src/pages/404.jsx:3]: Rollback this change, I didn't see any change

File: src/utils/styles.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import { makeStyles } from '@material-ui/core/styles';
```
Comment (ming-tsai) [inline src/utils/styles.js:1]: Replace all double quotes to single quote on this file

File: src/pages/index.jsx [hunk 3]
```diff
@@ -41,14 +38,14 @@ export default function Home({ data }) {
...
-          </Grid>
-        </Container>
+          <Container className={classes.cardGrid} maxWidth='xl'>
+            {/* End hero unit */}
+            <Grid container spacing={8}>
+              {posts.map((post, index) => (
+                <CardGrid item key={index} post={post} xs={12} sm={6} md={3} />
+              ))}
+            </Grid>
+          </Container>
```
Comment (ming-tsai) [inline src/pages/index.jsx:48]: Pleas fix the ident of these nodes

**Review by ming-tsai — CHANGES_REQUESTED** [review]

File: chinese/great-wall/begin-journey.md [hunk 4]
```diff
@@ -1,3 +1,3 @@
-你現在是一個小小兵，請不要選錯否則會被你的將軍砍頭或回家吃自己。
+你現在是一個小小兵，請不要選錯否則會被你的將軍砍頭或回家吃自己开始。
```
Comment (ming-tsai) [inline chinese/great-wall/begin-journey.md:1]: Why do you change this phrase?

**Comment by gauravsingh5** [comment]
The size of the cards didn't match because the description of the Chinese dungeon was 3 characters less. I used google translate and added "and start" to the description but you can change it as i don't know Chinese. Just add 3 more characters to the description.

**Review by ming-tsai — CHANGES_REQUESTED** [review]

**Comment by ming-tsai** [comment]
> The size of the cards didn't match because the description of the Chinese dungeon was 3 characters less. I used google translate and added "and start" to the description but you can change it as i don't know Chinese. Just add 3 more characters to the description.

Couldn't add the same size for the card o material ui?

**Comment by gauravsingh5** [comment]
I don't know about material ui much but let me try.

**Comment by ming-tsai** [comment]
Could you tell me what do you want to change?
Because for me the old card size is good, it doesn't seem too big or too small

**Comment by gauravsingh5** [comment]
All the cards were located in center and the space on the side was looking weird.

**Comment by ming-tsai** [comment]
For me is good, I'm just think have a background image 
image

**Comment by gauravsingh5** [comment]
Should we have 4 per row?

**Comment by ming-tsai** [comment]
We could try, but keep it the same size

**Comment by gauravsingh5** [comment]
Ok

**Comment by gauravsingh5** [comment]
image

**Comment by gauravsingh5** [comment]
Do you know how to size it?

**Comment by gauravsingh5** [comment]
image
I just sliced the excerpt

**Comment by gauravsingh5** [comment]
Maybe we can add 3 dots at the end.

**Comment by ming-tsai** [comment]
For me is good

File: src/components/card-grid.jsx [hunk 5]
```diff
@@ -25,25 +26,30 @@ export default function CardGrid(props) {
...
+          <CardMedia
+            className={classes.cardMedia}
+            image={defaultImage}
+            title={info.display}
+          />
+          <CardContent className={classes.cardContent}>
+            <Typography gutterBottom variant='h5' component='h2'>
+              {info.display}
+            </Typography>
+            <Typography>{post.excerpt.slice(0, 65)}</Typography>
```
Comment (ming-tsai) [inline src/components/card-grid.jsx:44]: Why just 65 characters?

**Comment by gauravsingh5** [comment]
It seemed to work

**Comment by gauravsingh5** [comment]
This is a temporary fix btw, we need to find a permanent one

File: src/components/card-grid.jsx (hunk 5 above)
Comment (ming-tsai) [reply src/components/card-grid.jsx:44]: If you just want 65 characters, you could use the gatsby graphql query, on the `pruneLength` change to 65
 image

**Comment by ming-tsai** [comment]
It fails because you change the text alignment to the center?

**Comment by gauravsingh5** [comment]
It's the number of lines it takes

**Comment by gauravsingh5** [comment]
image
This is right align

**Comment by gauravsingh5** [comment]
It's without the slice btw.

**Comment by ming-tsai** [comment]
> image
> This is right align

You could specify the `pruneLength` to another length
image

**Comment by ming-tsai** [comment]
But the question, why the current is align?
image

**Comment by gauravsingh5** [comment]
image
Prune 30 in English vs Chinese

**Comment by gauravsingh5** [comment]
image
This is Prune 10

**Comment by gauravsingh5** [comment]
I guess prune 10 should work.

**Comment by ming-tsai** [comment]
I will check your branch

**Comment by gauravsingh5** [comment]
Just check the [deploy preview](https://deploy-preview-144--markdown-dungeon.netlify.app)

**Comment by ming-tsai** [comment]
The 10 prune didn't show the description of dungeon

**Comment by gauravsingh5** [comment]
We can leave it like a teaser, if they click on it they see the full description.

**Comment by ming-tsai** [comment]
Let me check what is going with the card

**Comment by ming-tsai** [comment]
Hi @gauravsingh5,
Could you set the prune length to 100? I will check when merge to master

**Review by ming-tsai — APPROVED** [review]
Nice work 🎉🙌🙌🙌

**Pull request merged**